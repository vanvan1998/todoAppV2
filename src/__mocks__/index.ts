const todo = {
  email: 'nguyensivan1998+102@gmail.com',
  id: '82zB2At6gnxs0SbDJ8SW',
  notification: true,
  createdAt: '2024-05-10T01:14:38',
  title: '333333',
  detail: '3333333',
  startTime: '01:00:21',
  startDate: '05/10/2024',
  category: 1,
  completed: false,
  userId: 'j3WklyoTrBg4E1GKNz0Gn0az34V2',
  checkList: []
};
export { todo };
