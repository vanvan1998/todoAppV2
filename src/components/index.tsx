export { AddTodoModal } from './AddTodoModal';
export { Search } from './Search';
export { TodoItem } from './TodoItem';
export { Container } from './Container';
export { Header } from './Header';
export { AddTodo } from './addTodo';
export { SortBy } from './SortBy';
export { Input } from './input';
export { Button } from './button';
