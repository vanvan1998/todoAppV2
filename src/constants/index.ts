export const SORT_BY = {
  NEWEST: 'NEWEST',
  OLDEST: 'OLDEST',
  NONE: 'NONE'
};

export const MODE = {
  UPDATE: 'UPDATE',
  ADD: 'ADD'
};

export const DATE_FORMAT = 'MM/DD/YYYY';
export const TIME_FORMAT = 'HH:mm:ss';
