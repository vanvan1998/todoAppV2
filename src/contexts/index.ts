export * from './AppProvider';
export * from './GlobalLoadingProvider';
export * from './AuthProvider';
