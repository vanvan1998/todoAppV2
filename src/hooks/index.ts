export * from './useFirebase';
export * from './useNotification';
export * from './useMediaQuery';
