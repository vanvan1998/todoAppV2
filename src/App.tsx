import React from 'react';
import { AuthProvider } from './contexts/AuthContext';
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import {
  Signup,
  LeftMenu,
  Login,
  PrivateRoute,
  ForgotPassword,
  UpdatePassword,
  UpdateProfile,
  DashBoard
} from './modules';

function App() {
  return (
    <div style={{ minHeight: '100vh', display: 'flex' }}>
      <Router>
        <AuthProvider>
          <Routes>
            <Route
              path='/update-password'
              element={
                <PrivateRoute>
                  <UpdatePassword />
                </PrivateRoute>
              }
            />
            <Route
              path='/update-profile'
              element={
                <PrivateRoute>
                  <UpdateProfile />
                </PrivateRoute>
              }
            />
            <Route path='/signup' element={<Signup />} />
            <Route path='/login' element={<Login />} />
            <Route path='/forgot-password' element={<ForgotPassword />} />
            <Route
              path='/left-menu'
              element={
                <PrivateRoute>
                  <LeftMenu />
                </PrivateRoute>
              }
            />
            <Route
              path='/'
              element={
                <PrivateRoute>
                  <DashBoard />
                </PrivateRoute>
              }
            />
          </Routes>
        </AuthProvider>
      </Router>
    </div>
  );
}

export default App;
