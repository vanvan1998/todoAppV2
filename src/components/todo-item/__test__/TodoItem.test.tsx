import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import { TodoItem } from '..';
import { todo } from 'src/__mocks__';

jest.mock('next/navigation', () => {
  return {
    useRouter: jest.fn().mockReturnValue({
      push: jest.fn()
    })
  };
});

describe('Test TodoItem component', () => {
  test.only('should match its snapshot', async () => {
    const component = render(
      <TodoItem
        todo={todo}
        handleCompleteTodo={jest.fn()}
        handleDeleteTodo={jest.fn()}
        handleUpdateTodo={jest.fn()}
        onDragEnd={jest.fn()}
        onDragStart={jest.fn()}
      />
    );

    expect(component).toMatchSnapshot();
  });

  test.only('should render completed todo snapshot', async () => {
    const component = render(
      <TodoItem
        todo={{
          ...todo,
          notification: false,
          createdAt: '2024-05-10T01:14:38',
          startTime: '01:00:21',
          startDate: '05/10/2026',
          category: 2,
          completed: true,
          userId: 'j3WklyoTrBg4E1GKNz0Gn0az34V2'
        }}
        handleCompleteTodo={jest.fn()}
        handleDeleteTodo={jest.fn()}
        handleUpdateTodo={jest.fn()}
        onDragEnd={jest.fn()}
        onDragStart={jest.fn()}
      />
    );

    const button = component.getByTestId('task-details-button');
    fireEvent.click(button);

    const completeButton = component.getByTestId('complete-button');
    fireEvent.click(completeButton);

    const editButton = component.getByTestId('edit-button');
    fireEvent.click(editButton);
    fireEvent.click(editButton);

    const deleteButton = component.getByTestId('delete-button');
    fireEvent.click(deleteButton);
    expect(component).toMatchSnapshot();
  });
});
