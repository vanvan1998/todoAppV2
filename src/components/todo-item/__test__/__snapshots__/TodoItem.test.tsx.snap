// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TodoItem component should match its snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bALWzn lNTDe"
        draggable="true"
        id="82zB2At6gnxs0SbDJ8SW"
      >
        <div
          class="sc-gueXAH gRHGwY"
          style="font-weight: 500; padding-bottom: 6px;"
        >
          333333
        </div>
        <div
          class="sc-lnAfUH koPQRt"
        >
          <div
            class="sc-gueXAH gRHGwY"
            style="font-size: 12px; color: red;"
          >
            Expired time: 
          </div>
          <div
            class="sc-gueXAH gRHGwY"
            style="font-weight: bold; font-size: 12px; color: red;"
          >
            05/10/2024
             
            01:00:21
          </div>
        </div>
        <div
          class="sc-hLseSp fDeqZQ"
          style="height: 60px; width: 100%; margin-bottom: 8px;"
        >
          3333333
        </div>
        <div
          class="sc-ktEKgv jZWMWx"
        >
          <div
            class="sc-hAtFlw dfuuuD"
          >
            <div
              class="sc-gueXAH gRHGwY"
              style="color: rgb(210, 131, 18);"
            >
              Progress
            </div>
            <div
              class="sc-fmSBHF kyktUN"
              color="#d28312"
            >
              0
              %
            </div>
          </div>
          <div
            class="sc-fGFvMH kzrFFa"
            color="#d28312"
          >
            <div
              class="sc-blLrKi bOozld"
              color="#d28312"
              count="0"
            />
          </div>
        </div>
        <div
          class="sc-ivnBVK eMkHXj"
        >
          <button
            buttontype="primary"
            class="sc-gLDymS dItXMH"
            data-testid="task-details-button"
            style="min-height: 26px; height: 26px; padding: 0px; background-color: transparent; color: rgb(100, 89, 227);"
            type="button"
          >
            Task details
          </button>
          <div
            class="sc-cyRfdC kUlieG"
          >
            <button
              buttontype="primary"
              class="sc-gLDymS dItXMH"
              data-testid="complete-button"
              style="min-height: 26px; height: 26px; padding: 0px; background-color: transparent;"
              type="button"
            >
              <svg
                color="#6459e3"
                fill="currentColor"
                height="24"
                stroke="currentColor"
                stroke-width="0"
                version="1.1"
                viewBox="0 0 16 16"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 0c-4.418 0-8 3.582-8 8s3.582 8 8 8 8-3.582 8-8-3.582-8-8-8zM8 14c-3.314 0-6-2.686-6-6s2.686-6 6-6c3.314 0 6 2.686 6 6s-2.686 6-6 6z"
                />
              </svg>
            </button>
            <button
              buttontype="primary"
              class="sc-gLDymS dItXMH"
              data-testid="edit-button"
              style="background-color: white; padding: 0px 8px; border: 1px solid #6459e3; min-height: 26px; height: 26px; display: flex; align-items: center;"
              type="button"
            >
              <svg
                color="#6459e3"
                fill="currentColor"
                height="16"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M19.045 7.401c.378-.378.586-.88.586-1.414s-.208-1.036-.586-1.414l-1.586-1.586c-.378-.378-.88-.586-1.414-.586s-1.036.208-1.413.585L4 13.585V18h4.413L19.045 7.401zm-3-3 1.587 1.585-1.59 1.584-1.586-1.585 1.589-1.584zM6 16v-1.585l7.04-7.018 1.586 1.586L7.587 16H6zm-2 4h16v2H4z"
                />
              </svg>
            </button>
            <button
              buttontype="primary"
              class="sc-gLDymS dItXMH"
              data-testid="delete-button"
              style="background-color: white; padding: 0px 8px; border: 1px solid #6459e3; min-height: 26px; height: 26px; display: flex; align-items: center;"
              type="button"
            >
              <svg
                color="#6459e3"
                fill="currentColor"
                height="16"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M17 6H22V8H20V21C20 21.5523 19.5523 22 19 22H5C4.44772 22 4 21.5523 4 21V8H2V6H7V3C7 2.44772 7.44772 2 8 2H16C16.5523 2 17 2.44772 17 3V6ZM18 8H6V20H18V8ZM13.4142 13.9997L15.182 15.7675L13.7678 17.1817L12 15.4139L10.2322 17.1817L8.81802 15.7675L10.5858 13.9997L8.81802 12.232L10.2322 10.8178L12 12.5855L13.7678 10.8178L15.182 12.232L13.4142 13.9997ZM9 4V6H15V4H9Z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bALWzn lNTDe"
      draggable="true"
      id="82zB2At6gnxs0SbDJ8SW"
    >
      <div
        class="sc-gueXAH gRHGwY"
        style="font-weight: 500; padding-bottom: 6px;"
      >
        333333
      </div>
      <div
        class="sc-lnAfUH koPQRt"
      >
        <div
          class="sc-gueXAH gRHGwY"
          style="font-size: 12px; color: red;"
        >
          Expired time: 
        </div>
        <div
          class="sc-gueXAH gRHGwY"
          style="font-weight: bold; font-size: 12px; color: red;"
        >
          05/10/2024
           
          01:00:21
        </div>
      </div>
      <div
        class="sc-hLseSp fDeqZQ"
        style="height: 60px; width: 100%; margin-bottom: 8px;"
      >
        3333333
      </div>
      <div
        class="sc-ktEKgv jZWMWx"
      >
        <div
          class="sc-hAtFlw dfuuuD"
        >
          <div
            class="sc-gueXAH gRHGwY"
            style="color: rgb(210, 131, 18);"
          >
            Progress
          </div>
          <div
            class="sc-fmSBHF kyktUN"
            color="#d28312"
          >
            0
            %
          </div>
        </div>
        <div
          class="sc-fGFvMH kzrFFa"
          color="#d28312"
        >
          <div
            class="sc-blLrKi bOozld"
            color="#d28312"
            count="0"
          />
        </div>
      </div>
      <div
        class="sc-ivnBVK eMkHXj"
      >
        <button
          buttontype="primary"
          class="sc-gLDymS dItXMH"
          data-testid="task-details-button"
          style="min-height: 26px; height: 26px; padding: 0px; background-color: transparent; color: rgb(100, 89, 227);"
          type="button"
        >
          Task details
        </button>
        <div
          class="sc-cyRfdC kUlieG"
        >
          <button
            buttontype="primary"
            class="sc-gLDymS dItXMH"
            data-testid="complete-button"
            style="min-height: 26px; height: 26px; padding: 0px; background-color: transparent;"
            type="button"
          >
            <svg
              color="#6459e3"
              fill="currentColor"
              height="24"
              stroke="currentColor"
              stroke-width="0"
              version="1.1"
              viewBox="0 0 16 16"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 0c-4.418 0-8 3.582-8 8s3.582 8 8 8 8-3.582 8-8-3.582-8-8-8zM8 14c-3.314 0-6-2.686-6-6s2.686-6 6-6c3.314 0 6 2.686 6 6s-2.686 6-6 6z"
              />
            </svg>
          </button>
          <button
            buttontype="primary"
            class="sc-gLDymS dItXMH"
            data-testid="edit-button"
            style="background-color: white; padding: 0px 8px; border: 1px solid #6459e3; min-height: 26px; height: 26px; display: flex; align-items: center;"
            type="button"
          >
            <svg
              color="#6459e3"
              fill="currentColor"
              height="16"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19.045 7.401c.378-.378.586-.88.586-1.414s-.208-1.036-.586-1.414l-1.586-1.586c-.378-.378-.88-.586-1.414-.586s-1.036.208-1.413.585L4 13.585V18h4.413L19.045 7.401zm-3-3 1.587 1.585-1.59 1.584-1.586-1.585 1.589-1.584zM6 16v-1.585l7.04-7.018 1.586 1.586L7.587 16H6zm-2 4h16v2H4z"
              />
            </svg>
          </button>
          <button
            buttontype="primary"
            class="sc-gLDymS dItXMH"
            data-testid="delete-button"
            style="background-color: white; padding: 0px 8px; border: 1px solid #6459e3; min-height: 26px; height: 26px; display: flex; align-items: center;"
            type="button"
          >
            <svg
              color="#6459e3"
              fill="currentColor"
              height="16"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17 6H22V8H20V21C20 21.5523 19.5523 22 19 22H5C4.44772 22 4 21.5523 4 21V8H2V6H7V3C7 2.44772 7.44772 2 8 2H16C16.5523 2 17 2.44772 17 3V6ZM18 8H6V20H18V8ZM13.4142 13.9997L15.182 15.7675L13.7678 17.1817L12 15.4139L10.2322 17.1817L8.81802 15.7675L10.5858 13.9997L8.81802 12.232L10.2322 10.8178L12 12.5855L13.7678 10.8178L15.182 12.232L13.4142 13.9997ZM9 4V6H15V4H9Z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test TodoItem component should render completed todo snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    class="modal-open"
    data-rr-ui-modal-open=""
    style="padding-right: 1024px;"
  >
    <div>
      <div
        class="sc-bALWzn lNTDe"
        draggable="true"
        id="82zB2At6gnxs0SbDJ8SW"
      >
        <div
          class="sc-gueXAH gRHGwY"
          style="font-weight: 500; padding-bottom: 6px;"
        >
          333333
        </div>
        <div
          class="sc-lnAfUH koPQRt"
        >
          <div
            class="sc-gueXAH gRHGwY"
            style="font-size: 12px; color: green;"
          >
            Expired time: 
          </div>
          <div
            class="sc-gueXAH gRHGwY"
            style="font-weight: bold; font-size: 12px; color: green;"
          >
            05/10/2026
             
            01:00:21
          </div>
        </div>
        <div
          class="sc-hLseSp fDeqZQ"
          style="height: 60px; width: 100%; margin-bottom: 8px;"
        >
          3333333
        </div>
        <div
          class="sc-ktEKgv jZWMWx"
        >
          <div
            class="sc-hAtFlw dfuuuD"
          >
            <div
              class="sc-gueXAH gRHGwY"
              style="color: rgb(219, 15, 37);"
            >
              Progress
            </div>
            <div
              class="sc-fmSBHF eOLhQU"
              color="#db0f25"
            >
              100
              %
            </div>
          </div>
          <div
            class="sc-fGFvMH giKZNz"
            color="#db0f25"
          >
            <div
              class="sc-blLrKi cnEHvZ"
              color="#db0f25"
              count="100"
            />
          </div>
        </div>
        <div
          class="sc-ivnBVK eMkHXj"
        >
          <button
            buttontype="primary"
            class="sc-gLDymS dItXMH"
            data-testid="task-details-button"
            style="min-height: 26px; height: 26px; padding: 0px; background-color: transparent; color: rgb(100, 89, 227);"
            type="button"
          >
            Task details
          </button>
          <div
            class="sc-cyRfdC kUlieG"
          >
            <button
              buttontype="primary"
              class="sc-gLDymS dItXMH"
              data-testid="complete-button"
              style="min-height: 26px; height: 26px; padding: 0px; background-color: transparent;"
              type="button"
            >
              <svg
                color="#6459e3"
                fill="currentColor"
                height="24"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM369 209L241 337c-9.4 9.4-24.6 9.4-33.9 0l-64-64c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0l47 47L335 175c9.4-9.4 24.6-9.4 33.9 0s9.4 24.6 0 33.9z"
                />
              </svg>
            </button>
            <button
              buttontype="primary"
              class="sc-gLDymS dItXMH"
              data-testid="edit-button"
              style="background-color: white; padding: 0px 8px; border: 1px solid #6459e3; min-height: 26px; height: 26px; display: flex; align-items: center;"
              type="button"
            >
              <svg
                color="#6459e3"
                fill="currentColor"
                height="16"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M19.045 7.401c.378-.378.586-.88.586-1.414s-.208-1.036-.586-1.414l-1.586-1.586c-.378-.378-.88-.586-1.414-.586s-1.036.208-1.413.585L4 13.585V18h4.413L19.045 7.401zm-3-3 1.587 1.585-1.59 1.584-1.586-1.585 1.589-1.584zM6 16v-1.585l7.04-7.018 1.586 1.586L7.587 16H6zm-2 4h16v2H4z"
                />
              </svg>
            </button>
            <button
              buttontype="primary"
              class="sc-gLDymS dItXMH"
              data-testid="delete-button"
              style="background-color: white; padding: 0px 8px; border: 1px solid #6459e3; min-height: 26px; height: 26px; display: flex; align-items: center;"
              type="button"
            >
              <svg
                color="#6459e3"
                fill="currentColor"
                height="16"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M17 6H22V8H20V21C20 21.5523 19.5523 22 19 22H5C4.44772 22 4 21.5523 4 21V8H2V6H7V3C7 2.44772 7.44772 2 8 2H16C16.5523 2 17 2.44772 17 3V6ZM18 8H6V20H18V8ZM13.4142 13.9997L15.182 15.7675L13.7678 17.1817L12 15.4139L10.2322 17.1817L8.81802 15.7675L10.5858 13.9997L8.81802 12.232L10.2322 10.8178L12 12.5855L13.7678 10.8178L15.182 12.232L13.4142 13.9997ZM9 4V6H15V4H9Z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="fade modal-backdrop"
    />
    <div
      aria-modal="true"
      class="fade modal"
      role="dialog"
      style="width: 100%; height: 100vh; padding-top: 20px; display: block; padding-right: 0px;"
      tabindex="-1"
    >
      <div
        class="modal-dialog"
      >
        <div
          class="modal-content"
        >
          <div
            class="sc-fsQjcA kapxdL"
          >
            <div
              class="sc-qRuZU lcwWSj"
            >
              <div
                class="sc-gueXAH hWXdbp"
                styles="font-size: 20px; font-weight: 600;"
              >
                Update Task Todo
              </div>
              <div
                class="sc-jsMaUZ kfzPNd"
              />
            </div>
            <div
              style="margin-bottom: 16px;"
            >
              <div
                class="sc-gueXAH gRHGwY"
                style="font-size: 14px; font-weight: 500;"
              >
                Title Task
              </div>
              <input
                aria-label="Title Task"
                class="sc-iAEzLF gTKcTi"
                placeholder="Enter new task todo..."
                style="min-height: 40px; min-width: 438px;"
                type="text"
                value="333333"
              />
            </div>
            <div
              class="sc-gueXAH bAvcGT"
              styles="font-weight: 500; margin-bottom: 8px;"
            >
              Category
            </div>
            <div
              class="sc-kFuvnu eCHfHO"
            >
              <button
                buttontype="primary"
                class="sc-gLDymS dItXMH"
                data-testid="upcoming-button"
                style="height: 36px; min-height: 36px; padding: 0px 16px; display: flex; align-items: center; gap: 8px; width: 130px; background-color: rgb(182, 180, 185);"
                type="button"
              >
                <svg
                  fill="currentColor"
                  height="16"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 448 512"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V192H0v272zm64-192c0-8.8 7.2-16 16-16h96c8.8 0 16 7.2 16 16v96c0 8.8-7.2 16-16 16H80c-8.8 0-16-7.2-16-16v-96zM400 64h-48V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H160V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H48C21.5 64 0 85.5 0 112v48h448v-48c0-26.5-21.5-48-48-48z"
                  />
                </svg>
                Upcoming
              </button>
              <button
                buttontype="primary"
                class="sc-gLDymS dItXMH"
                data-testid="important-button"
                style="height: 36px; min-height: 36px; padding: 0px 16px; display: flex; align-items: center; gap: 8px; width: 130px; background-color: rgb(182, 180, 185);"
                type="button"
              >
                <svg
                  color="#d28312"
                  fill="currentColor"
                  height="16"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10.01 21.01c0 1.1.89 1.99 1.99 1.99s1.99-.89 1.99-1.99h-3.98zM12 6c2.76 0 5 2.24 5 5v7H7v-7c0-2.76 2.24-5 5-5zm0-4.5c-.83 0-1.5.67-1.5 1.5v1.17C7.36 4.85 5 7.65 5 11v6l-2 2v1h18v-1l-2-2v-6c0-3.35-2.36-6.15-5.5-6.83V3c0-.83-.67-1.5-1.5-1.5zM11 8h2v4h-2zm0 6h2v2h-2z"
                  />
                </svg>
                Important
              </button>
              <button
                buttontype="primary"
                class="sc-gLDymS dItXMH"
                data-testid="urgent-button"
                style="height: 36px; min-height: 36px; padding: 0px 16px; display: flex; align-items: center; gap: 8px; width: 130px; background-color: rgb(100, 89, 227);"
                type="button"
              >
                <svg
                  color="#db0f25"
                  fill="currentColor"
                  height="16"
                  stroke="currentColor"
                  stroke-width="0"
                  style="transform: rotate(270deg);"
                  viewBox="0 0 24 24"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                  <path
                    d="m3.5 18.99 11 .01c.67 0 1.27-.33 1.63-.84L20.5 12l-4.37-6.16c-.36-.51-.96-.84-1.63-.84l-11 .01L8.34 12 3.5 18.99z"
                  />
                </svg>
                Urgent
              </button>
            </div>
            <div
              class="sc-gueXAH bAvcGT"
              styles="font-weight: 500; margin-bottom: 8px;"
            >
              Description
            </div>
            <textarea
              class="sc-fbJSV liFuaL"
              placeholder="Enter description..."
            >
              3333333
            </textarea>
            <div
              class="sc-irTrJD dQTfHZ"
            >
              <div
                class="sc-gueXAH hmBKDZ"
                styles="font-size: 14px; font-weight: 500;"
              >
                Notification
              </div>
              <div
                class="form-check form-switch"
              >
                <input
                  class="form-check-input"
                  id="flexSwitchCheckDefault"
                  role="switch"
                  style="width: 40px; height: 20px;"
                  type="checkbox"
                />
              </div>
            </div>
            <div
              class="sc-brKelq eBpLhd"
            >
              <button
                buttontype="secondary"
                class="sc-gLDymS jcdnJc"
                data-testid="cancel-todo-button"
                style="flex: 1;"
                type="button"
              >
                Cancel
              </button>
              <button
                buttontype="primary"
                class="sc-gLDymS dItXMH"
                data-testid="add-todo-button"
                style="flex: 1;"
                type="button"
              >
                Update
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bALWzn lNTDe"
      draggable="true"
      id="82zB2At6gnxs0SbDJ8SW"
    >
      <div
        class="sc-gueXAH gRHGwY"
        style="font-weight: 500; padding-bottom: 6px;"
      >
        333333
      </div>
      <div
        class="sc-lnAfUH koPQRt"
      >
        <div
          class="sc-gueXAH gRHGwY"
          style="font-size: 12px; color: green;"
        >
          Expired time: 
        </div>
        <div
          class="sc-gueXAH gRHGwY"
          style="font-weight: bold; font-size: 12px; color: green;"
        >
          05/10/2026
           
          01:00:21
        </div>
      </div>
      <div
        class="sc-hLseSp fDeqZQ"
        style="height: 60px; width: 100%; margin-bottom: 8px;"
      >
        3333333
      </div>
      <div
        class="sc-ktEKgv jZWMWx"
      >
        <div
          class="sc-hAtFlw dfuuuD"
        >
          <div
            class="sc-gueXAH gRHGwY"
            style="color: rgb(219, 15, 37);"
          >
            Progress
          </div>
          <div
            class="sc-fmSBHF eOLhQU"
            color="#db0f25"
          >
            100
            %
          </div>
        </div>
        <div
          class="sc-fGFvMH giKZNz"
          color="#db0f25"
        >
          <div
            class="sc-blLrKi cnEHvZ"
            color="#db0f25"
            count="100"
          />
        </div>
      </div>
      <div
        class="sc-ivnBVK eMkHXj"
      >
        <button
          buttontype="primary"
          class="sc-gLDymS dItXMH"
          data-testid="task-details-button"
          style="min-height: 26px; height: 26px; padding: 0px; background-color: transparent; color: rgb(100, 89, 227);"
          type="button"
        >
          Task details
        </button>
        <div
          class="sc-cyRfdC kUlieG"
        >
          <button
            buttontype="primary"
            class="sc-gLDymS dItXMH"
            data-testid="complete-button"
            style="min-height: 26px; height: 26px; padding: 0px; background-color: transparent;"
            type="button"
          >
            <svg
              color="#6459e3"
              fill="currentColor"
              height="24"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM369 209L241 337c-9.4 9.4-24.6 9.4-33.9 0l-64-64c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0l47 47L335 175c9.4-9.4 24.6-9.4 33.9 0s9.4 24.6 0 33.9z"
              />
            </svg>
          </button>
          <button
            buttontype="primary"
            class="sc-gLDymS dItXMH"
            data-testid="edit-button"
            style="background-color: white; padding: 0px 8px; border: 1px solid #6459e3; min-height: 26px; height: 26px; display: flex; align-items: center;"
            type="button"
          >
            <svg
              color="#6459e3"
              fill="currentColor"
              height="16"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19.045 7.401c.378-.378.586-.88.586-1.414s-.208-1.036-.586-1.414l-1.586-1.586c-.378-.378-.88-.586-1.414-.586s-1.036.208-1.413.585L4 13.585V18h4.413L19.045 7.401zm-3-3 1.587 1.585-1.59 1.584-1.586-1.585 1.589-1.584zM6 16v-1.585l7.04-7.018 1.586 1.586L7.587 16H6zm-2 4h16v2H4z"
              />
            </svg>
          </button>
          <button
            buttontype="primary"
            class="sc-gLDymS dItXMH"
            data-testid="delete-button"
            style="background-color: white; padding: 0px 8px; border: 1px solid #6459e3; min-height: 26px; height: 26px; display: flex; align-items: center;"
            type="button"
          >
            <svg
              color="#6459e3"
              fill="currentColor"
              height="16"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17 6H22V8H20V21C20 21.5523 19.5523 22 19 22H5C4.44772 22 4 21.5523 4 21V8H2V6H7V3C7 2.44772 7.44772 2 8 2H16C16.5523 2 17 2.44772 17 3V6ZM18 8H6V20H18V8ZM13.4142 13.9997L15.182 15.7675L13.7678 17.1817L12 15.4139L10.2322 17.1817L8.81802 15.7675L10.5858 13.9997L8.81802 12.232L10.2322 10.8178L12 12.5855L13.7678 10.8178L15.182 12.232L13.4142 13.9997ZM9 4V6H15V4H9Z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
